### PR TITLE
Webpack configured

### PR DESCRIPTION
**Pull request by AllaShtatska**

### CODE REVIEW

File: client/.gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/dist
+
 /node_modules
 
 package-lock.json
```
Comment (nakhodkin) [inline client/.gitignore:5]: _trivial_
I think, that the general purpose of keeping lock files for project dependencies in VCS is to allow everyone(your colleagues, CI😁) to have exactly the same dependencies tree as you had while you were working on the project. So it is actually good practice to keep lock files in the repo with a code.

_hint_
If you don't want to read lock file when installing dependencies you can specify it e.g.
`$ yarn install --no-lockfile` - for `yarn` package manager, I believe that the similar one exists for `npm` as well

File: client/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+
+module.exports = function (env, options) {
+  const isProduction = options.mode === "production";
+
+  const config = {
+    context: path.resolve(__dirname, "src"),
+
+    mode: isProduction ? "production" : "development",
```
Comment (nakhodkin) [inline client/webpack.config.js:10]: _trivial_
I wonder whether the `mode` option is set automatically in regard with specifying `--mode production`.

I assume that the task for homework was about setting `--env.production`
Screen Shot 2019-03-20 at 2 02 38 AM

If you ask me what's the difference, I believe that it is about that fact `mode` is automatically saturate your configuration with some bunch of predefined settings, as it plugs in `optimizations` for a prod bundle also turns on minification for `.js` files and so on(the more detailed list of these options you can find here: https://webpack.js.org/concepts/mode/

So basically if you want to customize `webpack.config` manually without using any of "presets" like that come in with `mode` option, you can try using `--env` option.

But I'll check my assumption and tell you why they chose `--env` option in favour of `--mode`

File: client/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    devtool: isProduction ? "none" : "source-map",
```
Comment (nakhodkin) [inline client/webpack.config.js:12]: _trivial_
Actually, there was any requirement about specifying a particular value for `devtool` option, but it really comes extremely handy to specify source-map, especially for production hence you've got minified source-code in prod bundle and source-maps, allow you to navigate, debug and keep track of execution of your code with no difference to you whether it non-readable minified prod code or source code for development.

File: client/package.json [hunk 4]
```diff
@@ -1,9 +1,30 @@
 {
-  "name": "server",
+  "name": "client",
   "version": "0.0.0",
   "private": true,
+  "scripts": {
+    "build:prod": "webpack --mode production",
+    "build:dev": "webpack --mode development",
+    "run:prod": "webpack-dev-server --mode production",
+    "run:dev": "webpack-dev-server --mode development",
```
Comment (nakhodkin) [inline client/package.json:9]: _trivial_
I was thinking about what if we want to startup both web servers concurrently😁, we might need to specify the `port` option also.

File: client/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    plugins: [
+      new HtmlWebpackPlugin({
+        hash: true,
+        template: "./index.html"
+      })
+    ]
+  };
+
+  return config;
```
Comment (AllaShtatska) [inline client/webpack.config.js]: comment
